code/modules/crafting/guncrafting.dm

```diff
@@ -24,66 +24,101 @@
 
 /obj/item/weaponcrafting/gunkit/ebow
 	name = "\improper energy crossbow parts kit"
 	desc = "Highly illegal weapons refurbishment kit that allows you to turn a laser gun into a near-duplicate energy crossbow. Almost like the real thing!"
 	origin_tech = "combat=4;magnets=4;syndicate=2"
+	outcome = /obj/item/gun/energy/kinetic_accelerator/crossbow/large
 
 /obj/item/weaponcrafting/gunkit/immolator
 	name = "\improper immolator laser gun parts kit"
 	desc = "Take a perfectly functioning laser gun. Butcher the inside of the gun so it runs hot and mean. You now have a immolator laser. You monster."
 	origin_tech = "combat=4;magnets=4;powerstorage=3"
+	outcome = /obj/item/gun/energy/immolator
 
 /obj/item/weaponcrafting/gunkit/accelerator
 	name = "\improper accelerator laser cannon parts kit"
 	desc = "A suitcase containing the necessary gun parts to transform a standard laser gun into an accelerator laser cannon."
 	origin_tech = "combat=4;magnets=4;powerstorage=3"
+	outcome = /obj/item/gun/energy/lasercannon
 
 /obj/item/weaponcrafting/gunkit/lwap
 	name = "\improper lwap laser sniper parts kit"
 	desc = "A suitcase containing the necessary gun parts to transform an laser gun into an advanced piercing laser sniper. Now with wall hacks!"
 	origin_tech = "combat=6;magnets=6;powerstorage=4"
+	outcome = /obj/item/gun/energy/lwap
 
 /obj/item/weaponcrafting/gunkit/plasma
 	name = "\improper plasma pistol parts kit"
 	desc = "A suitcase containing the necessary gun parts to transform a standard laser gun into a plasma pistol. Wort, wort, wort!"
 	origin_tech = "combat=4;magnets=4;powerstorage=3"
+	outcome = /obj/item/gun/energy/plasma_pistol
 
 /obj/item/weaponcrafting/gunkit/u_ionsilencer
 	name = "\improper u-ion silencer parts kit"
 	desc = "A suitcase containing the necessary gun parts to transform a standard disabler into a silenced and lethal disabling weapon. Look officer, he has no wounds from me!"
 	origin_tech = "combat=6;magnets=6;syndicate=2"
+	outcome = /obj/item/gun/energy/disabler/silencer
+
+/obj/item/weaponcrafting/gunkit/universal_gun_kit
+	name = "\improper universal self assembling gun parts kit"
+	desc = "A suitcase containing the necessary gun parts to build a full gun, when combined with a gun kit. Use it directly on a gunkit to rapidly assemble it."
+	icon_state = "syndicase"
+
+/obj/item/weaponcrafting/gunkit/universal_gun_kit/afterattack(atom/movable/AM, mob/user, flag)
+	if(istype(AM, /obj/item/weaponcrafting/gunkit))
+		var/obj/item/weaponcrafting/gunkit/G = AM
+		if(!G.outcome)
+			to_chat(user, "<span class='warning'>That gunkit can not be used to craft a weapon.</span>")
+			return
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		if(!do_after(user, 5 SECONDS, target = user))
+			return
+		playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+		var/obj/item/I = new G.outcome
+		user.unEquip(src)
+		user.put_in_hands(I)
+		qdel(G)
+		qdel(src)
```

Comment:
Changes suggested:
More descriptive var name(s), guard clause, general simplification of assignment code
```suggestion
/obj/item/weaponcrafting/gunkit/universal_gun_kit/afterattack(obj/item/weaponcrafting/gunkit/gunkit_to_use, mob/user, flag)
	if(!istype(gunkit_to_use))
		return
	if(!gunkit_to_use.outcome)
		to_chat(user, "<span class='warning'>That gunkit can not be used to craft a weapon.</span>")
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	if(!do_after(user, 5 SECONDS, target = user))
		return
	playsound(user, 'sound/items/drill_use.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
	var/obj/item/gun_produced = new gunkit_to_use.outcome
	user.unEquip(src)
	user.put_in_hands(gun_produced)
	qdel(gunkit_to_use)
	qdel(src)
```